### PR TITLE
Change userList paging from "by frontend" to "by backend"

### PR DESCRIPTION
**Pull request by AllForNothing**
Change userList paging from "by frontend" to "by backend"
#8173
Signed-off-by: AllForNothing <sshijun@vmware.com>

### CODE REVIEW

File: src/portal/src/app/user/user.component.ts [hunk 1]
```diff
@@ -41,348 +47,303 @@ import { throwError as observableThrowError } from "rxjs";
...
-    let observableLists: any[] = [];
-    if (users && users.length) {
-      users.forEach(user => {
-        observableLists.push(this.delOperate(user));
-      });
-
-      forkJoin(...observableLists).subscribe((item) => {
+
+    // Refresh the user list
+    refreshUser(from: number, to: number): void {
```
Comment (jwangyangls) [inline src/portal/src/app/user/user.component.ts]: where do use the params ?

File: src/portal/src/app/user/user.component.ts [hunk 2]
```diff
@@ -41,348 +47,303 @@ import { throwError as observableThrowError } from "rxjs";
...
+        this.refresh();
+    }
 
-  SelectedChange(): void {
-    this.forceRefreshView(5000);
-  }
+    // Data loading
+    load(state: any): void {
+        this.selectedRow = [];
+        this.getUserListByPaging();
```
Comment (jwangyangls) [inline src/portal/src/app/user/user.component.ts]: where loading?

File: src/portal/src/app/user/user.component.ts [hunk 3]
```diff
@@ -41,348 +47,303 @@ import { throwError as observableThrowError } from "rxjs";
...
 
-  openChangePwdModal(): void {
-    if (this.selectedRow.length === 1) {
-      this.changePwdDialog.open(this.selectedRow[0].user_id);
+    // Filter items by keywords
+    doFilter(terms: string): void {
+        this.selectedRow = [];
+        this.currentTerm = terms.trim();
+        this.currentPage = 1;
+        this.getUserListByPaging();
```
Comment (jwangyangls) [inline src/portal/src/app/user/user.component.ts]: where loading?

File: src/portal/src/app/user/user.component.ts (hunk 1 above)
Comment (AllForNothing) [reply src/portal/src/app/user/user.component.ts]: fixed

File: src/portal/src/app/user/user.component.ts (hunk 2 above)
Comment (AllForNothing) [reply src/portal/src/app/user/user.component.ts]: fixed

File: src/portal/src/app/user/user.component.ts (hunk 3 above)
Comment (AllForNothing) [reply src/portal/src/app/user/user.component.ts]: fixed

**Pull request merged**